showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/services/ItemService.java

```diff
@@ -20,14 +19,32 @@
         return itemDao.findAll();
     }
 
-    public Item findByItemID(Integer itemID) {
-        Optional<Item> item =itemDao.findById(itemID);
-        try{
-            item.get();
-        }catch(NoSuchElementException ex){
-            System.out.println("No user with that ID");
+    public Item addItem(Item item) {
+        return itemDao.saveAndFlush(item);
+    }
+
+    public void deleteItem(Integer id) {
+        itemDao.deleteById(id);
+    }
+
+    public Item getItem(Integer id) {
+
+        return itemDao.findById(id).get();
+    }
+
+    public Item editItem(Item item, Integer id) {
+        Item current = itemDao.getOne(id);
+        BeanUtils.copyProperties(item,current,"id");
+        return itemDao.saveAndFlush(current);
+    }
+    //Used to reset the database for testing
+    //If theres a more efficient way let me know =) -Jesse
+    public void deleteAllRows() {
+        List<Item> items = this.getAllItems();
+        for (int i = 0;i<items.size();++i) {
+            Integer id = items.get(i).getId();
+            this.deleteItem(id);
         }
-        return item.get();
     }
```

Comment:
would you be able to just use <repoName>.deleteAll()?
https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#deleteAll--

not experienced with JPA though, so feel free to let me know if you've already tried this.

also, i would just take out the comments before merging the final version of this to the staging branch. i think if you're looking for some code optimizations, you can just feel free to tag us on the specific lines of code you're thinking about when you send up your PR